utils/utils.py

```diff
@@ -0,0 +1,173 @@
+        input_field.send_keys(Keys.TAB)
+    except NoSuchElementException:
+        raise NoSuchElementException(f"Element with {selector_type} '{selector_value}' not found.")
+
+
+def universal_wait_for(driver, condition_type, selector_type=None, selector_value=None, different_value=None,
+                       timeout=get_variable_value("TIMEOUT")):
+    """
+    Uniwersalna funkcja oczekiwania, która czeka na spełnienie różnych warunków.
+
+    Args:
+    driver (WebDriver): Instancja WebDrivera używana do interakcji z przeglądarką.
+    condition_type (callable): Typ warunku z modułu expected_conditions.
+    selector_type (By): Typ selektora (np. By.ID, By.CSS_SELECTOR).
+    selector_value (str): Wartość selektora.
+    timeout (int): Maksymalny czas oczekiwania w sekundach (domyślnie 10).
+
+    Returns:
+    WebElement: Znaleziony element, jeśli warunek został spełniony.
+
+    Raises:
+    TimeoutException: Jeśli warunek nie zostanie spełniony w określonym czasie.
+    """
+    try:
+        if selector_value is not None and selector_type is not None:
+            element = WebDriverWait(driver, timeout).until(
+                condition_type((selector_type, selector_value))
+            )
+        else:
+            element = WebDriverWait(driver, timeout).until(
+                condition_type(different_value)
+            )
+        return element
+    except TimeoutException:
+        raise TimeoutException(
+            f"Condition {condition_type.__name__} not met for element with {selector_type} '{selector_value}' within {timeout} seconds.")
+
+
+def get_text_from_elements_by_class(driver, selector_type, selector_value):
+    """
+    Pobiera wartości tekstowe z elementów HTML o podanej klasie.
+
+    Args:
+    driver (webdriver): Instancja WebDriver używana do kontroli przeglądarki.
+    class_name (str): Nazwa klasy elementów, z których ma być pobierany tekst.
+
+    Returns:
+    List[str]: Lista zawierająca wartości tekstowe z elementów o podanej klasie.
+    """
+    # Znajdowanie elementów po klasie
+    elements = driver.find_elements(selector_type, selector_value)
+
+    # Pobieranie tekstów z każdego elementu
+    texts = [element.text for element in elements]
+
+    return texts
+
+
+def generate_login_data():
+    fake = Faker('pl_PL')
+    first_name = fake.first_name()
+    last_name = fake.last_name()
+    login = unidecode(first_name).lower()[:3] + unidecode(last_name).lower()[:3]
+    email = fake.email()
+    phone_number = fake.phone_number().replace(' ', '')[-9:]
+    birth_day = fake.date_of_birth().strftime('%d.%m.%Y')
+    password = get_variable_value("TEST_PASSWORD")
+
+    return LoginData(first_name, last_name, login, email, phone_number, birth_day, password)
+
+
+if __name__ == "__main__":
+
+    def print_all_env_variables():
+        print("All variables:")
+        for key, value in get_all_variables():
+            print(f"{key}: {value}")
+
+
+    print_all_env_variables()
```

Comment:
To już można wywalić, to było tylko takie pokazowe. No chyba że wciąż tego potrzebujecie